modules/services/pizauth.nix

```
@@ -0,0 +1,195 @@
+              default = [ ];
+              example = [
+                "https://outlook.office365.com/IMAP.AccessAsUser.All"
+                "https://outlook.office365.com/SMTP.Send"
+                "offline_access"
+              ];
+            };
+
+            loginHint = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = ''
+                The login hint for the account provider. Optional.
+              '';
+            };
+
+            extraConfig = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = ''
+                Additional configuration that will be added to the account configuration. See pizauth.conf(5) for available options.
+              '';
+            };
+          };
+        }
+      );
+      default = { };
+      description = "Pizauth accounts that should be configured";
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [ (lib.hm.assertions.assertPlatform "services.pizauth" pkgs lib.platforms.linux) ];
+
+    home.packages = [ cfg.package ];
+
+    xdg.configFile."pizauth.conf".source =
+      let
+        renderScopes =
+          scopes:
+          let
+            indent = "  ";
+            quoted = map (s: "\"${s}\"") scopes;
+            joined = lib.concatStringsSep ",\n${indent}${indent}" quoted;
+          in
+          "[\n${indent}${indent}${joined}\n${indent}]";
+
+        renderAccount =
+          let
+            indent = "  ";
+          in
+          name: acc:
+          ''
+            account "${name}" {
+            ${indent}auth_uri = "${acc.authUri}";
+            ${indent}token_uri = "${acc.tokenUri}";
+            ${indent}client_id = "${acc.clientId}";
+            ${indent}client_secret = "${acc.clientSecret}";
+          ''
+          + lib.optionalString (acc.scopes != [ ] && acc.scopes != null) ''
+            ${indent}scopes = ${renderScopes acc.scopes};
+          ''
+          + lib.optionalString (acc.loginHint != "" && acc.loginHint != null) ''
+            ${indent}login_hint = "${acc.loginHint}";
+          ''
+          + lib.optionalString (acc.extraConfig != "" && acc.extraConfig != null) (
+            let
+              indentedExtraConfig = lib.concatMapStringsSep "\n" (
+                line: if line == "" then "" else "${indent}${line}"
+              ) (lib.splitString "\n" acc.extraConfig);
+            in
+            indentedExtraConfig
+          )
+          + ''
+            }
+          '';
+
+        renderedAccounts = lib.concatStringsSep "\n" (
+          lib.mapAttrsToList (name: acc: renderAccount name acc) cfg.accounts
+        );
+
+        configFile = pkgs.writeTextFile {
+          name = "pizauth.conf";
+          text =
+            lib.optionalString (cfg.extraConfig != null && cfg.extraConfig != "") "${cfg.extraConfig}\n"
+            + renderedAccounts;
+        };
+      in
+      configFile;
```

```suggestion
      in
      pkgs.writeTextFile {
        name = "pizauth.conf";
        text =
          lib.optionalString (cfg.extraConfig != null && cfg.extraConfig != "") "${cfg.extraConfig}\n"
          + renderedAccounts;
      };
```

Can just reduce noise by not storing the final step in a variable just to be referenced by name immediately.